### DISCUSSION TITLE
How to solve the problem of sprite rendering order?

### DISCUSSION THREAD

**Discussion by PoisonousGame** [godotengine/godot-proposals#4491] [2D]
How to solve the problem of sprite rendering order?
The sword and shield are separated into two sprites, and I need to change the order in which they are displayed in the animation. If I use z_index, the following problem will occur. Sword and shield are hidden behind trees. Since I need Ysort to sort according to the Y coordinate, and I want the sword and shield to only be rendered and sorted relative to the inside of the character, and they are at the same z-index for the outside, how to solve this problem?
![动画](https://user-images.githubusercontent.com/39994989/166628190-7d0d4d05-7ea0-4438-9b20-e2a9fdae0a6f.gif)
![Screenshot 2022-05-04 131653](https://user-images.githubusercontent.com/39994989/166628201-fd1720fd-da04-4e4a-923f-7b2c9fe3dacc.png)
![Screenshot 2022-05-04 151707](https://user-images.githubusercontent.com/39994989/166638399-549d81c4-5d1f-4391-9887-228439c85895.png)

The character (KinematicBody2D3) has 3 sprites, two of which are sword and shield. The sword and shield are located behind the cha sprite at the beginning. I want to play the sword drawing animation, and the sword sprite is located in front of the cha sprite. This effect can be achieved with z_index. However, when the character is placed on the map and moved, due to the z_index rendering problem, the sword and shield are hidden behind the tree.

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/60748.
